service/controller/v19/adapter/guest_subnets.go

```diff
@@ -1,63 +1,44 @@
 package adapter
 
 import (
-	"net"
+	"fmt"
 
 	"github.com/giantswarm/aws-operator/service/controller/v19/key"
-	"github.com/giantswarm/ipam"
-	"github.com/giantswarm/microerror"
 )
 
+type Subnet struct {
+	Index               int
+	AvailabilityZone    string
+	CIDR                string
+	Name                string
+	MapPublicIPOnLaunch bool
+}
+
 type GuestSubnetsAdapter struct {
-	PublicSubnetAZ                   string
-	PublicSubnetCIDR                 string
-	PublicSubnetName                 string
-	PublicSubnetMapPublicIPOnLaunch  bool
-	PrivateSubnetAZ                  string
-	PrivateSubnetCIDR                string
-	PrivateSubnetName                string
-	PrivateSubnetMapPublicIPOnLaunch bool
+	PublicSubnets  []Subnet
+	PrivateSubnets []Subnet
 }
 
 func (s *GuestSubnetsAdapter) Adapt(cfg Config) error {
-	publicSubnet, privateSubnet, err := allocatePublicAndPrivateSubnets(cfg)
-	if err != nil {
-		return microerror.Mask(err)
+	for i, az := range key.AvailabilityZones(cfg.CustomObject) {
+		snet := Subnet{
+			Index:               i,
+			AvailabilityZone:    az.Name,
+			CIDR:                az.Subnet.Public.CIDR,
+			Name:                fmt.Sprintf("PublicSubnet%d", i),
```

Comment:
```suggestion
			Name:                fmt.Sprintf("PublicSubnet%02d", i),
```

I think it's a bit cooler when they will have the same length.